### PR TITLE
Compute features

### PR DESCRIPTION
**Pull request by AlexanderHHansen**
Created files to add features to our dataframe. We compute features like min, max, median, mean for each trip for a segment and also aggregates for each of them. We refactored the script df_processing which computed speeds and distances, to be part of compute_features folder. Thus, tests were moved

### CODE REVIEW

File: tests/preprocessing/compute_features/test_compute_features.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "func, speed_column, expected",
+    [
+        (min, [[23, 42, 82], [47, 58, 60]], [23, 47]),
+        (max, [[0, 47, 88], [89, 200, 67]], [88, 200]),
+        (mean, [[1, 2], [4, 4]], [1.5, 4]),
+        (median, [[1, 2, 3, 4, 5]], [3]),
+    ],
+)
+
+# TODO: check where none/empty lists in coloumns should be handled
```
Comment (JakobFaarGreg) [inline tests/preprocessing/compute_features/test_compute_features.py:35]: Er det her ikke prioriteret for denne pull request?

**Review by JakobFaarGreg — CHANGES_REQUESTED** [review]

File: archive/analysis_metrics.py [hunk 2]
```diff
@@ -132,7 +127,7 @@ def main():
     df = create_df_from_json(filename)
     university_boulevard_osm_id = 10240935
     filtered_df = filter_segments(df, university_boulevard_osm_id)
-    calculate_distance_and_speed(filtered_df)
+    # calculate_distance_and_speed(filtered_df)
```
Comment (anesheim-uni) [inline archive/analysis_metrics.py:130]: Remove if not needed.

File: pipeline/preprocessing/compute_features/calculate_speeds_distances.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
+# append root to path
+import sys
+from os.path import dirname, realpath
+
+current_dir = dirname(realpath(__file__))
+parentdir = dirname(dirname(dirname(current_dir)))
+sys.path.append(parentdir)
```
Comment (anesheim-uni) [inline pipeline/preprocessing/compute_features/calculate_speeds_distances.py:7]: Er dette nødvendigt nu når vi har fixed imports?

File: pipeline/preprocessing/compute_features/calculate_speeds_distances.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+from math import sqrt
+import pandas as pd  # type: ignore
+from pipeline.preprocessing.compute_features.type_alias import Trip
+
+coordinate = tuple[float, float]  # type alias
+
+
+def calc_utm_dist(utm1: coordinate, utm2: coordinate) -> float:
+    """calculate the distance between two utm coordinates
+    uses formula for euclidian distance, which is not 100% accurate?
```
Comment (anesheim-uni) [inline pipeline/preprocessing/compute_features/calculate_speeds_distances.py:19]: "Which is not 100 accurate?" skal vel ikke med?

File: pipeline/preprocessing/compute_features/type_alias.py [hunk 5]
```diff
@@ -0,0 +1,8 @@
+from typing import Any, Callable
+import pandas as pd  # type: ignore
+
+UnixTimestamp = int
+CoordinateWithTime = tuple[float, float, UnixTimestamp]
+Trip = list[CoordinateWithTime]
+ListOfSpeeds = list[list[float]]
+FeatureCalculator = Callable[[pd.DataFrame], Any]
```
Comment (anesheim-uni) [inline pipeline/preprocessing/compute_features/type_alias.py:8]: Disse kunne godt bruges flere steder i compute_features.

File: pipeline/preprocessing/compute_features/compute_features.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+
+def aggregate_results(
+    aggregate_func: Callable, feature: Feature, row: pd.DataFrame
+) -> Union[None, float]:
+    """
+    Aggregate results in a given feature coloumn with input aggregate function (e.g. mean).
+
+    Args:
+        aggregate_func: the function to a
```
[the remark this answers is not in the document]
Comment (AlexanderHHansen) [reply pipeline/preprocessing/compute_features/compute_features.py:54]: rettet

File: archive/analysis_metrics.py (hunk 2 above)
Comment (AlexanderHHansen) [reply archive/analysis_metrics.py:130]: har ikke noget med pr at gøre

File: tests/preprocessing/compute_features/test_calculate_speeds_distances.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
+import sys
+from os.path import dirname, realpath
+
+current_dir = dirname(realpath(__file__))
+parentdir = dirname(dirname(dirname(current_dir)))
+sys.path.append(parentdir)
```
Comment (anesheim-uni) [inline tests/preprocessing/compute_features/test_calculate_speeds_distances.py:6]: Er det ikke unødvendigt jf. import PR?

**Review by anesheim-uni — COMMENTED** [review]
Minor details/changes, pretty good overall

File: tests/preprocessing/compute_features/test_calculate_speeds_distances.py (hunk 7 above)
Comment (AlexanderHHansen) [reply tests/preprocessing/compute_features/test_calculate_speeds_distances.py:6]: Ting failer hvis det ikke er med. Bliver dermed fixet i imports pr.

**Review by anesheim-uni — APPROVED** [review]
All good

**Review by JakobFaarGreg — APPROVED** [review]
lgtm

**Pull request merged**